pkg/controller/endpoint/endpoints_controller_test.go

```diff
@@ -1207,209 +1204,152 @@ func TestPodToEndpointAddressForService(t *testing.T) {
-
-			enableDualStack: false,
-			ipFamilies:      ipv4ipv6,
-
+			name:       "v4 legacy headless service, in a dual stack cluster",
+			ipFamilies: ipv4ipv6,
 			service: v1.Service{
 				Spec: v1.ServiceSpec{
 					ClusterIP: v1.ClusterIPNone,
 				},
 			},
-
 			expectedEndpointFamily: ipv4,
 		},
 		{
-			name: "v4 legacy headless service, in a dual stack ipv6-primary cluster",
-
-			enableDualStack: true,
-			ipFamilies:      ipv6ipv4,
-
+			name:       "v4 legacy headless service, in a dual stack ipv6-primary cluster",
+			ipFamilies: ipv6ipv4,
 			service: v1.Service{
 				Spec: v1.ServiceSpec{
 					ClusterIP: v1.ClusterIPNone,
 				},
 			},
-
 			expectedEndpointFamily: ipv6,
 		},
 		{
-			name: "v6 service, in a dual stack cluster",
-
-			enableDualStack: true,
-			ipFamilies:      ipv4ipv6,
-
+			name:       "v6 service, in a dual stack cluster",
+			ipFamilies: ipv4ipv6,
 			service: v1.Service{
 				Spec: v1.ServiceSpec{
 					ClusterIP: "3000::1",
 				},
 			},
-
 			expectedEndpointFamily: ipv6,
 		},
 		{
-			name: "v6 headless service, in a single stack cluster",
-
-			enableDualStack: false,
-			ipFamilies:      ipv6only,
-
+			name:       "v6 headless service, in a single stack cluster",
+			ipFamilies: ipv6only,
 			service: v1.Service{
 				Spec: v1.ServiceSpec{
 					ClusterIP: v1.ClusterIPNone,
 				},
 			},
-
 			expectedEndpointFamily: ipv6,
 		},
 		{
-			name: "v6 headless service, in a dual stack cluster (connected to a new api-server)",
-
-			enableDualStack: true,
-			ipFamilies:      ipv4ipv6,
-
+			name:       "v6 headless service, in a dual stack cluster (connected to a new api-server)",
+			ipFamilies: ipv4ipv6,
 			service: v1.Service{
 				Spec: v1.ServiceSpec{
 					ClusterIP:  v1.ClusterIPNone,
 					IPFamilies: []v1.IPFamily{v1.IPv6Protocol}, // <- set by a api-server defaulting logic
 				},
 			},
-
 			expectedEndpointFamily: ipv6,
 		},
 		{
-			name: "v6 legacy headless service, in a dual stack cluster  (connected to a old api-server)",
-
-			enableDualStack: false,
-			ipFamilies:      ipv4ipv6,
-
+			name:       "v6 legacy headless service, in a dual stack cluster  (connected to a old api-server)",
+			ipFamilies: ipv4ipv6,
 			service: v1.Service{
 				Spec: v1.ServiceSpec{
 					ClusterIP: v1.ClusterIPNone, // <- families are not set by api-server
 				},
 			},
-
 			expectedEndpointFamily: ipv4,
 		},
-
 		// in reality this is a misconfigured cluster
 		// i.e user is not using dual stack and have PodIP == v4 and ServiceIP==v6
 		// we are testing that we will keep producing the expected behavior
 		{
-			name: "v6 service, in a v4 only cluster. dual stack disabled",
-
-			enableDualStack: false,
-			ipFamilies:      ipv4only,
-
+			name:       "v6 service, in a v4 only cluster. dual stack disabled",
+			ipFamilies: ipv4only,
 			service: v1.Service{
 				Spec: v1.ServiceSpec{
 					ClusterIP: "3000::1",
 				},
 			},
-
+			expectError:            true,
```

Comment:
The comment on this case says "we are testing that we will keep producing the expected behavior", but you're changing the expected result.

Reply:
I have modified the comment. The test is correct and should expect error. Previously we just returned PodIP blindly. now we check if it matches the family of the service.